data-access/Cargo.toml

```diff
@@ -16,8 +16,8 @@
 # under the License.
 
 [package]
-name = "datafusion-storage"
-description = "Storage for DataFusion query engine"
+name = "data-access"
```

Comment:
`data-access` feel a bit too generic to me. To be more specific, how about we name this crate and the folder `object-store`? the object-store crate hasn't been claimed in crates.io yet.

Reply:
i had the idea that within this other data access providers / crates could exist - for example maybe stream provider or database provider and thats why the name was generic.

Reply:
As @matthewmturner mentioned, the object store may be too specific and maybe will limit the extension of future data access abilities.